scripts/export-script.py

```diff
@@ -0,0 +1,94 @@
+import sys
+import boto3
+import re
+from awsglue.transforms import *
+from awsglue.utils import getResolvedOptions
+from pyspark.context import SparkContext
+from awsglue.context import GlueContext
+from awsglue.job import Job
+from awsglue.dynamicframe import DynamicFrame
+from datetime import datetime
+
+glueContext = GlueContext(SparkContext.getOrCreate())
+job = Job(glueContext)
+
+args = getResolvedOptions(sys.argv, ['JOB_NAME', 'jobId', 'exportType', 'transactionTime', 'since', 'outputFormat', 'glueDatabase', 'glueTableName', 's3OutputBucket'])
+
+# type and groupId are optional parameters
+type = None
+if ('--{}'.format('type') in sys.argv):
+    type = getResolvedOptions(sys.argv, ['type'])['type']
+groupId = None
+if ('--{}'.format('groupId') in sys.argv):
+    groupId = getResolvedOptions(sys.argv, ['groupId'])['groupId']
+
+job_id = args['jobId']
+export_type = args['exportType']
+transaction_time = args['transactionTime']
+since = args['since']
+outputFormat = args['outputFormat']
+glue_table_name = args['glueTableName']
+
+glue_database = args['glueDatabase']
+bucket_name = args['s3OutputBucket']
+
+# Read data from DDB
+original_data_source_dyn_frame = glueContext.create_dynamic_frame.from_catalog(database = glue_database, table_name = glue_table_name)
```

Comment:
Were you testing this script on the Glue console?
Can you add the annotations like `## @type: DataSource` so that it generates the execution plan diagram? 

https://docs.aws.amazon.com/glue/latest/dg/edit-script.html

Reply:
For most of the development of the script, I used `sagemaker` to develop and test the script. [Link here](https://aws.amazon.com/about-aws/whats-new/2018/10/aws-glue-now-supports-connecting-amazon-sagemaker-notebooks-to-development-endpoints/)

One the script is working in Sagemaker I ported it to the Glue console for the final test. I don't think the annotations are strictly necessary. It's more helpful if you're starting out from a Glue template and are making minor changes. The annotation is then helpful if you want to visually see it using Glue's diagram.

For more complex scripts, I think comments in the code is enough. For example, the [sample glue scripts](https://github.com/aws-samples/aws-glue-samples/tree/master/examples) released by AWS doesn't have the diagram annotations.